spinalcordtoolbox/reports/qc.py

```diff
@@ -264,85 +283,130 @@ def wrapped_f(sct_slice, *args):
             """
             self.qc_report.slice_name = sct_slice.get_name()
 
-            # Get the aspect ratio (height/width) based on pixel size. Consider only the first 2 slices.
-            aspect_img, self.aspect_mask = sct_slice.aspect()[:2]
-
             self.qc_report.make_content_path()
             logger.info('QcImage: %s with %s slice', func.__name__, sct_slice.get_name())
 
-            if self._angle_line is None:
-                img, mask = func(sct_slice, *args)
-            else:
-                [img, mask], centermass = func(sct_slice, *args)
+            # Get the aspect ratio (height/width) based on pixel size. Consider only the first 2 slices.
+            self.aspect_img, self.aspect_mask = sct_slice.aspect()[:2]
+
+            if self.process in ['sct_fmri_moco', 'sct_dmri_moco']:
+                [images_after_moco, images_before_moco], centermass = func(sct_slice, *args)
                 self._centermass = centermass
+                self._make_QC_image_for_4d_volumes(images_after_moco, images_before_moco)
+
+            else:
+                if self._angle_line is not None:
+                    [img, mask], centermass = func(sct_slice, *args)
+                    self._centermass = centermass
+                else:
+                    img, mask = func(sct_slice, *args)
```

Comment:
I'm quite new to git, is there a way to "remove" changes like this one from the PR, where the only difference is that the code has been moved a few lines?

Reply:
And about 'self._centermass = centermass', it needs to be called before '_make_QC_image_for_4d_volumes' and '_make_QC_image_for_3d_volumes', so I cannot really move it outside!

Reply:
> And about 'self._centermass = centermass', it needs to be called before '_make_QC_image_for_4d_volumes' and '_make_QC_image_for_3d_volumes', so I cannot really move it outside!

Oh, of course! I was mistaken, you're absolutely right. :sweat_smile:

Reply:
> I'm quite new to git, is there a way to "remove" changes like this one from the PR, where the only difference is that the code has been moved a few lines?

If you were working on a team that cared about having a perfectly neat/tidy commit history for PRs, then what you could do is make a commit for the change, then use [`git rebase -i`](https://thoughtbot.com/blog/git-interactive-rebase-squash-amend-rewriting-history#interactive-rebase) along with the ["fixup"](https://thoughtbot.com/blog/git-interactive-rebase-squash-amend-rewriting-history#squash-commits-together) option to combine the new commit with an older commit.

Here at SCT, though, when a PR is merged, all of the commits will get squashed (so that there is a single commit on master representing this PR). Because of that, there's not much benefit to rebasing, so it's OK to just create a new commit that undoes the change.

(One other reason not to use rebases for work that was already pushed is because it can make a PR a little confusing to read through. Because of how rebasing works, the edited commits become entirely new commits, so they will appear at the bottom of the PR, removed from the context of the original discussion.)

Reply:
> I'm quite new to git

Also, I would have never guessed! It's been very nice and easy to collaborate with you. :smiley:

Reply:
Thanks so much! And I definitely learned a lot! 😄